### PR TITLE
fix: update metadata on successfull end of update process (MAPCO-3528)

### PR DESCRIPTION
**Pull request by razbroc**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✔                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

### CODE REVIEW

File: src/jobs/models/jobsManager.ts [hunk 1]
```diff
@@ -254,6 +244,16 @@ export class JobsManager {
         // eslint-disable-next-line @typescript-eslint/no-magic-numbers
         await this.jobManager.updateJobById(job.id, OperationStatus.COMPLETED, 100, undefined, catalogRecord.id);
         job.status = OperationStatus.COMPLETED;
+
+        const mergedData = this.metadataMerger.merge(catalogRecord.metadata, job.metadata);
```
Comment (asafMasa) [inline src/jobs/models/jobsManager.ts:238]: before previous action (job update)

File: tests/unit/tasks/models/jobsModel.spec.ts [hunk 2]
```diff
@@ -299,8 +299,8 @@ describe('JobsManager', () => {
       await jobsManager.completeJob(jobId, taskId);
 
       expect(updateJobByIdMock).toHaveBeenCalledTimes(0);
-      expect(mergeMock).toHaveBeenCalledTimes(1);
-      expect(updateMock).toHaveBeenCalledTimes(1);
+      expect(mergeMock).toHaveBeenCalledTimes(0);
```
Comment (asafMasa) [inline tests/unit/tasks/models/jobsModel.spec.ts:302]: check if this is being checked on job completion

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍

File: tests/unit/tasks/models/jobsModel.spec.ts (hunk 2 above)
Comment (razbroc) [reply tests/unit/tasks/models/jobsModel.spec.ts:302]: on test `it('should complete job once all tasks are successful for update-merge job-task'`

Comment (razbroc) [reply tests/unit/tasks/models/jobsModel.spec.ts:302]: `      expect(updateMock).toHaveBeenCalledTimes(1);`

**Review by asafMasa — APPROVED** [review]

**Pull request merged**